pkg/cmd/infra/router/router.go

```diff
@@ -83,6 +95,48 @@ func (o *RouterSelection) RouteSelectionFunc() controller.RouteHostFunc {
 	}
 }
 
+func (o *RouterSelection) AdmissionCheck(route *routeapi.Route) error {
+	if len(route.Spec.Host) < 1 {
+		return nil
+	}
+
+	if hostInDomainList(route.Spec.Host, o.BlacklistedDomains) {
+		glog.V(4).Infof("host %s in list of denied domains", route.Spec.Host)
+		return fmt.Errorf("host in list of denied domains")
+	}
+
+	if o.WhitelistedDomains.Len() > 0 {
+		glog.V(4).Infof("Checking if host %s is in the list of allowed domains", route.Spec.Host)
+		if hostInDomainList(route.Spec.Host, o.WhitelistedDomains) {
+			glog.V(4).Infof("host %s admitted - in the list of allowed domains", route.Spec.Host)
+			return nil
+		}
+
+		glog.V(4).Infof("host %s rejected - not in the list of allowed domains", route.Spec.Host)
+		return fmt.Errorf("host not in the allowed list of domains")
+	}
+
+	glog.V(4).Infof("host %s admitted", route.Spec.Host)
+	return nil
+}
+
+// RouteAdmissionFunc returns a func that checks if a route can be admitted
+// based on blacklist & whitelist checks and wildcard routes policy setting.
+// Note: The blacklist settings trumps the whitelist ones.
+func (o *RouterSelection) RouteAdmissionFunc() controller.RouteAdmissionFunc {
+	return func(route *routeapi.Route) (error, bool) {
+		if err := o.AdmissionCheck(route); err != nil {
+			return err, false
+		}
+
+		if len(route.Spec.WildcardPolicy) > 0 && route.Spec.WildcardPolicy != routeapi.WildcardPolicyNone && !o.AllowWildcardRoutes {
```

Comment:
the old router didn't know about wildcards. the new one does, and is disabling them.

no different than a pod with initContainers running without init containers in 1.3, and being rejected in 1.4 if the initContainers are disallowed. the 1.4 server doesn't just skip the initContainers and run the main containers.